### PR TITLE
reduce overhead in each test case on riak_test

### PR DESCRIPTION
**Pull request by kuenishi**
- not restarting after specifying admin spec
- starting riak_cs and stanchion can run in parallel with ring converge
- change ring_creation_size to 8 from default

With https://github.com/basho/riak_test/pull/641 my whole riak_test execution from 36 minutes to 22 minutes by wallclock time.

after:

```
real    22m52.258s
user    5m9.256s
sys     2m23.628s
```

before:

```
real    36m10.808s
user    5m33.780s
sys     2m58.804s
```

This pull request also ties up several regression tests, to two test modules. This is for avoiding overhead of each harness setup/teardown.

### CODE REVIEW

File: riak_test/tests/regression_tests_2.erl [hunk 1]
```diff
@@ -17,20 +17,25 @@
...
               {cs, rtcs:cs_config([{fold_objects_for_list_keys, true}])}],
     {UserConfig, {_RiakNodes, _CSNodes, _Stanchion}} = rtcs:setup(2, Config),
+
+    verify_cs654(UserConfig),
+    cs_631_regression_test:verify_cs631(UserConfig),
+    cs_781_regression_test:run_test(UserConfig),
+    pass.
+
+%% @doc Integration test for [https://github.com/basho/riak_cs/issues/654]
+verify_cs654(UserConfig) ->
```
Comment (shino) [inline riak_test/tests/regression_tests_2.erl:72]: This test case is not so short, so feels like worth for dedicated module?

File: riak_test/tests/regression_tests.erl [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        case catch erlcloud_s3:list_objects(BucketName, UserConfig) of
+            {'EXIT', {{aws_error, Error2}, _}} ->
+                Error2;
+            Result2 ->
+                Result2
+        end,
+    ?assert(verify_error_response(ListObjectRes2)),
+    pass.
+
+verify_error_response({_, 404, _, RespStr}) ->
```
Comment (shino) [inline riak_test/tests/regression_tests.erl]: This function and blow seems general enough to move to `rtcs` or some helper module.

File: riak_test/tests/regression_tests_2.erl [hunk 3]
```diff
@@ -17,20 +17,25 @@
...
--export([confirm/0]).
-
 confirm() ->
     Config = [{riak, rtcs:riak_config()}, {stanchion, rtcs:stanchion_config()},
               {cs, rtcs:cs_config([{fold_objects_for_list_keys, true}])}],
     {UserConfig, {_RiakNodes, _CSNodes, _Stanchion}} = rtcs:setup(2, Config),
+
+    verify_cs654(UserConfig),
+    cs_631_regression_test:verify_cs631(UserConfig),
+    cs_781_regression_test:run_test(UserConfig),
```
Comment (shino) [inline riak_test/tests/regression_tests_2.erl]: This works well but just for better appearance, how do you like some rules for modules?
For example, for any `XXX`,

```
pass = cs_XXX_regression:verify(SetupInfo),
```

Comment (shino) [inline riak_test/tests/regression_tests_2.erl]: The above comment is for test cases with dedicated modules.
Ones that just functions in the module (such as `verify_cs654` at line 32) seems nice for me :)

**Comment by shino** [comment]
All r_t passes and much faster than before. Great!
Although small nitpicks, +1 for overall.

**Comment by kuenishi** [comment]
I did some cleanup, so could you check it once again, please?

**Comment by shino** [comment]
```
================ regression_tests_2 failure stack trace =====================
{{badmatch,ok},
 [{regression_tests_2,confirm,0,
                      [{file,"riak_test/tests/regression_tests_2.erl"},
                       {line,37}]}]}
=============================================================================
```

**Comment by kuenishi** [comment]
@borshop merge

**Pull request merged**